src/main/kotlin/org/wycliffeassociates/otter/jvm/app/ui/takemanagement/view/RecordableFragment.kt

```diff
@@ -0,0 +1,202 @@
+                    /* We can't just add the node being dragged, since the selected take might have just been
+                        loaded from the database */
+                    else -> add(createTakeCard(it))
+                }
+            }
+        }
+    }
+
+    protected val dragComponents = DragComponents()
+    private val dragTargetTop = dragComponents.dragTargetTop()
+
+    private val dragContainer = VBox().apply {
+        draggingNodeProperty.onChange {
+            clear()
+            it?.let { node -> add(node) }
+        }
+    }
+
+    init {
+        importStylesheet<AppStyles>()
+        createAudioPluginProgressDialog()
+    }
+
+    final override val root: Parent = anchorpane {
+        addDragTakeEventHandlers()
+        addButtonEventHandlers()
+
+        createSnackBar(this)
+
+        add(mainContainer.apply {
+            anchorpaneConstraints {
+                leftAnchor = 0.0
+                rightAnchor = 0.0
+                bottomAnchor = 0.0
+                topAnchor = 0.0
+            }
+        })
+        add(dragContainer)
+    }
+
+    private fun Parent.addDragTakeEventHandlers() {
+        addEventHandler(StartDragEvent.START_DRAG, ::startDrag)
+        addEventHandler(AnimateDragEvent.ANIMATE_DRAG, ::animateDrag)
+        addEventHandler(CompleteDragEvent.COMPLETE_DRAG, ::completeDrag)
+    }
+
+    private fun Parent.addButtonEventHandlers() {
+        addEventHandler(PlayOrPauseEvent.PLAY) {
+            lastPlayOrPauseEvent.set(it)
+        }
+        addEventHandler(DeleteTakeEvent.DELETE_TAKE) {
+            recordableViewModel.deleteTake(it.take)
+        }
+        addEventHandler(EditTakeEvent.EDIT_TAKE) {
+            recordableViewModel.editTake(it)
+        }
+    }
+
+    private fun createSnackBar(pane: Pane) {
+        // TODO: This doesn't actually handle anything correctly. Need to know whether the user
+        // TODO... hasn't selected an editor or recorder and respond appropriately.
+        val snackBar = JFXSnackbar(pane)
+        recordableViewModel.snackBarObservable.subscribe {
+            snackBar.enqueue(
+                JFXSnackbar.SnackbarEvent(
+                    messages["noRecorder"],
+                    messages["addPlugin"].toUpperCase(),
+                    5000,
+                    false,
+                    EventHandler {
+                        audioPluginViewModel.addPlugin(true, false)
+                    })
+            )
+        }
+    }
+
+    private fun createAudioPluginProgressDialog() {
+        // Plugin active cover
+        val dialog = progressdialog {
+            root.addClass(AppStyles.progressDialog)
+            recordableViewModel.contextProperty.toObservable().subscribe { newContext ->
+                when (newContext) {
+                    TakeContext.RECORD -> graphic = AppStyles.recordIcon("60px")
+                    TakeContext.EDIT_TAKES -> graphic = AppStyles.editIcon("60px")
+                    else -> {}
```

Comment:
Unfortunately when I remove else, I have to add a null case or else there is a compiler warning. I think this is still better than having else though in case we add more options to this enum, like you said before.